### PR TITLE
Add unittest for compression

### PR DESCRIPTION
**Pull request by karan6181**
## Description
- Add unittest for compression algorithms.

### CODE REVIEW

File: streaming/base/compression/compression.py [hunk 1]
```diff
@@ -132,7 +132,7 @@ class Snappy(Compression):
 
     extension = 'snappy'
 
-    def compress(self, data: bytes) -> bytes:
+    def compress(self, data: Union[bytes, str]) -> bytes:
```
Comment (karan6181) [inline streaming/base/compression/compression.py:135]: Looks like snappy compression algorithms takes string as well apart from bytes. Hence, made this changes.

Comment (knighton) [reply streaming/base/compression/compression.py:135]: we are not using this with strings for our purposes, let's not enable that use case here?

Comment (karan6181) [reply streaming/base/compression/compression.py:135]: Looks like python annotations is very strict. If we keep it as `bytes` and you pass a string, then it won't throw any error and at a same time, it compressed the data and return the data in `bytes` !! 😞

Comment (knighton) [reply streaming/base/compression/compression.py:135]: would rather keep it uniform with the other compression algo interfaces and call this silent extra capability an easter egg? up to you if you feel strongly about it

**Review by knighton — APPROVED** [review]

File: streaming/base/compression/compression.py (hunk 1 above)
Comment (karan6181) [reply streaming/base/compression/compression.py:135]: Cool. Let me change it back to `bytes` as this is not a major concern.

Comment (karan6181) [reply streaming/base/compression/compression.py]: Done in https://github.com/mosaicml/streaming/pull/25/commits/0e5d87e61ee2b027df73751826c615755ffdf499.

**Pull request merged**